rust/ql/src/queries/diagnostics/MissingElements.ql

```diff
@@ -39,23 +29,33 @@ string multipleString(int i) {
 query predicate listUnimplemented(string location, string msg) {
   // something that is not extracted yet
   exists(int c |
-    c = strictcount(Unimplemented n | cleanLocationString(getUnimplementedLocation(n)) = location) and
+    c =
+      strictcount(Unimplemented n |
+        exists(getUnimplementedLocation(n).getFile().getRelativePath()) and
```

Comment:
This excludes elements from user code for which the extractor did not know a location.

Reply:
The MissingElement query is a bit special. It's purpose is to measure the quality of the database/extractor. If we throw out all MissingElements without location then we might think a database is in perfect health while in reality it is not.

Reply:
... and we're committed to *not* extracting non-user code now as I understand it (still now 100% sure how that works).  I'll try including them again.

Reply:
I've just reverted the restriction to user code in this PR as well.  I'm fairly sure I was seeing inconsistencies with it though, we will see...

Reply:
Yep.
```
-| @0:0:0:0 | Not yet implemented (x20). |
+| @0:0:0:0 | Not yet implemented (x14). |
```
This is the difference between my local results and CI.  Is this something we can fix in the extractor quickly, or should I perhaps change the PR to align with CI (in which case we can expect local test failures until the issue is fixed)?

Reply:
I think I found the source of the trouble. Could you try removing the use of the `println!` macro? I suspect that macro gets expanded differently depending on the rust version/target platform, or whatever other difference in environment .

Reply:
That reduces me to "Not yet implemented (x17)" locally.

Reply:
Did you comment out the`println!` in `my_macro.rs` as well as `main.rs`?

Reply:
... oops ...

Yep, that gets me down to 14, which is presumably in line with CI now.  I've pushed the fix.  There *is* still an issue in the extractor presumably, but this PR should be mergeable now I think.

Reply:
Thanks, yes there must be some issue, or perhaps the standard library for rust on the CI runners has a different implementation for `println!`.

Reply:
hmm, that is unfortunate. But maybe we just need to do a https://github.com/actions-rust-lang/setup-rust-toolchain step for running the QL tests?